pkgs/development/libraries/librist/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, meson
+, ninja
+, cmake
+, pkg-config
+, cjson
+, cmocka
+, mbedtls
+}:
+
+stdenv.mkDerivation rec {
+  pname = "librist";
+  version = "0.2.6";
+
+  src = fetchFromGitLab {
+    domain = "code.videolan.org";
+    owner = "rist";
+    repo = "librist";
+    rev = "v${version}";
+    hash = "sha256-EgWEMRiBJGq7ILmRqdL7Ok6/FhZNmYgiYyWHUhTZqCA=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    cmake
```

Comment:
Are you sure we need both `meson` *and* `cmake` here?

Reply:
Without meson it doesn't build at all with: 

```
this derivation will be built:
  /nix/store/1mjrjlx7dpjjgc9sc38zkj0zd87999kk-librist-0.2.6.drv
building '/nix/store/1mjrjlx7dpjjgc9sc38zkj0zd87999kk-librist-0.2.6.drv'...
unpacking sources
unpacking source archive /nix/store/i4y48pa96akwiaikcj2alhh5ywww1n52-source
source root is source
patching sources
configuring
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/share/doc/librist -DCMAKE_INSTALL_INFODIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/include -DCMAKE_INSTALL_SBINDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/j061mvdxw70xz20r2r0lr5lk0rcsjf9i-gcc-wrapper-11.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/j061mvdxw70xz20r2r0lr5lk0rcsjf9i-gcc-wrapper-11.3.0/bin/ranlib -DCMAKE_AR=/nix/store/j061mvdxw70xz20r2r0lr5lk0rcsjf9i-gcc-wrapper-11.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/dn9zxgwv7dsq767rhwchq8jk5x5s7xly-librist-0.2.6
CMake Warning:
  Ignoring extra path from command line:

   ".."


CMake Error: The source directory "/build/source" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
```

without cmake it builds but the build-log doesn't look correct like with (see https://github.com/NixOS/nixpkgs/pull/194518#issuecomment-1268709378):
```
this derivation will be built:
  /nix/store/7x3mvn67ahlx3dddkmn28jzmvrbakls4-librist-0.2.6.drv
building '/nix/store/7x3mvn67ahlx3dddkmn28jzmvrbakls4-librist-0.2.6.drv'...
unpacking sources
unpacking source archive /nix/store/i4y48pa96akwiaikcj2alhh5ywww1n52-source
source root is source
patching sources
configuring
meson flags: --buildtype=plain         --libdir=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/lib --libexecdir=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/libexec         --bindir=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/bin --sbindir=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/sbin         --includedir=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/include         --mandir=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/share/man --infodir=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/share/info         --localedir=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/share/locale         -Dauto_features=enabled         -Dwrap_mode=nodownload         --prefix=/nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6
The Meson build system
Version: 0.63.1
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Project name: libRIST
Project version: 0.2.6
C compiler for the host machine: gcc (gcc 11.3.0 "gcc (GCC) 11.3.0")
C linker for the host machine: gcc ld.bfd 2.39
Host machine cpu family: x86_64
Host machine cpu: x86_64
Check usable header "linux/if_alg.h" : YES
Checking for function "clock_gettime" : YES
Compiler for C supports arguments -Wundef: YES
Compiler for C supports arguments -Werror=vla: YES
Compiler for C supports arguments -Wno-maybe-uninitialized: YES
Compiler for C supports arguments -Wno-missing-field-initializers: YES
Compiler for C supports arguments -Wno-unused-parameter: YES
Compiler for C supports arguments -Wshorten-64-to-32: NO
Compiler for C supports arguments -Wunused-parameter: YES
Compiler for C supports arguments -Wmaybe-uninitialized: YES
Compiler for C supports arguments -Wno-error=deprecated-declarations: YES
Run-time dependency threads found: YES
Compiler for C supports arguments -Watomic-implicit-seq-cst: NO
Compiler for C supports arguments -fvisibility=hidden: YES
Check usable header "stdatomic.h" : YES
Found pkg-config: /nix/store/p44bsik684s8n77drkihwx6551c6q189-pkg-config-wrapper-0.29.2/bin/pkg-config (0.29.2)
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency cjson found: NO (tried pkgconfig and cmake)
Has header "cjson/cJSON.h" : YES
Library cjson found: YES
Message: Building mbedtls
Run-time dependency mbedcrypto found: NO (tried pkgconfig and cmake)
Has header "mbedtls/aes.h" : YES
Library mbedcrypto found: YES
Configuring version.h using configuration
Program objcopy found: YES
Configuring config.h using configuration
Message: Building tools
Library cmocka found: YES
test/rist/meson.build:72: DEPRECATION: ":" is not allowed in test name "Main profile encryption receive server mode unencrupted, sender: client mode", it has been replaced with "_"
test/rist/meson.build:75: DEPRECATION: ":" is not allowed in test name "Main profile encryption client mode, sender: server mode unencrypted", it has been replaced with "_"
Build targets in project: 8
NOTICE: Future-deprecated features used:
 * 0.56.0: {'meson.source_root'}

libRIST 0.2.6

  User defined options
    auto_features: enabled
    bindir       : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/bin
    buildtype    : plain
    includedir   : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/include
    infodir      : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/share/info
    libdir       : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/lib
    libexecdir   : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/libexec
    localedir    : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/share/locale
    mandir       : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/share/man
    prefix       : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6
    sbindir      : /nix/store/p7mli8f7f7ly8xkahw7jp5pkfm212kxz-librist-0.2.6/sbin
    wrap_mode    : nodownload

Found ninja-1.11.1 at /nix/store/4q68zlkgl96wlvgn2pz0x1k8vdib983n-ninja-1.11.1/bin/ninja
meson: enabled parallel building
building
build flags: -j16 -l16
```

Reply:
Alright, fair enough. :)

Reply:
Would be nice if you could approve the PR if there is nothing more todo ✌🏻

Reply:
If the only difference is the runtime dependency then you can safely remove `cmake` from it. It doesn't matter to building (thus `runtime`).